src/main/scala/sigmastate/utils/SigmaByteReader.scala

```diff
@@ -1,16 +1,16 @@
 package sigmastate.utils
 
-import java.nio.ByteBuffer
-
-import scorex.util.serialization.{Reader, VLQByteBufferReader}
+import scorex.util.Extensions._
+import scorex.util.serialization.Reader
 import sigmastate.SType
 import sigmastate.Values.SValue
-import sigmastate.serialization.{ConstantStore, TypeSerializer, ValDefTypeStore, ValueSerializer}
-import scorex.util.Extensions._
+import sigmastate.lang.exceptions.DeserializeCallDepthExceeded
+import sigmastate.serialization.{ConstantStore, SigmaSerializer, TypeSerializer, ValDefTypeStore, ValueSerializer}
 
 class SigmaByteReader(val r: Reader,
                       var constantStore: ConstantStore,
-                      var resolvePlaceholdersToConstants: Boolean)
+                      var resolvePlaceholdersToConstants: Boolean,
```

Comment:
@greenhat do we really need this fields to be `var`?

Reply:
Yes, we're setting new values for existing reader (see https://github.com/ScorexFoundation/sigmastate-interpreter/blob/471450d25ca7e129b13193dafea7adfeca95f994/src/main/scala/sigmastate/serialization/ErgoTreeSerializer.scala#L33 )